specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,262 @@
+# Base Profile
+
+This profile defines a subset of the QIR specification to support a coherent set
+of functionalities and capabilities that might be offered by a system. The Base
+Profile specifies the minimal requirements for executing quantum programs.
+Execution of a Base Profile compliant program does not, for instance, require
+supporting quantum instructions dependent on measurement outcomes. The only
+requirements for a backend are to be able to perform unitary transformations on
+the quantum state, as well as measurements at the end of the program.
+
+The intention is for front ends to emit programs that comply with the QIR
+specification, and for a compilation stage to then target this QIR to comply
+with the Base Profile specification. This targeting is possible in principle if
+the program doesn't make use of any features that would require hardware
+support.
+
+To support execution of Base Profile compliant programs, a backend must have the
+following fundamental capabilities:
+
+1. It supports executing a sequence of quantum instructions that transforms the
+   quantum state.
+2. It supports measuring the state of each qubit at the end of a computation.
+3. It must be able to return the measured value for each qubit in the order
+   requested by the program. This can be done in software as a post processing
+   step after all quantum instructions and the final measurements have been
+   performed; it does not require hardware or runtime support.
```

Comment:
The intend is to spec a small set of possible output schemas a backend can choose from (currently two - one if output is generated async and order is not guaranteed and one that doesn't require string labels -, and I am not sure we will ever need a third one, but I am sure we will update the two). We are effectively asking that a frontend would be able to process either one (e.g. to present visuals instead of the rather verbose raw output), though the tooling to do so can be open and shared.